openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/ReportDataRepository.java

```diff
@@ -39,10 +39,28 @@ public void deleteReportDataAtDate(ReportDataType reportDataType, String date) {
     cleanUpIndex(reportDataType, date);
   }
 
+  public void deletePreviousReportData(ReportDataType reportDataType) {
+    if (reportDataType == ReportDataType.RAW_COST_ANALYSIS_REPORT_DATA) {
+      ((CollectionDAO.ReportDataTimeSeriesDAO) timeSeriesDao).deletePreviousReportDataType(reportDataType.value());
+      cleanUpPreviousIndex(reportDataType);
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid report data type value. Only Allowed value is "
+              + ReportDataType.RAW_COST_ANALYSIS_REPORT_DATA.value());
+    }
+  }
+
```

Comment:
I don't need that validation since you already define the schema in the resource with `@Schema(implementation = ReportDataType.class))`

Reply:
can we also rename the method to `deletePreviousData`?